pkg/cmos/cmos.go

```diff
@@ -16,18 +16,32 @@ const (
 	cmosDataPort = 0x71
 )
 
+type CMOSChip struct {
+	In  func(uint16, memio.UintN) error
+	Out func(uint16, memio.UintN) error
+}
+
 // Read reads a register reg from CMOS into data.
-func Read(reg memio.Uint8, data memio.UintN) error {
-	if err := memio.Out(cmosRegPort, &reg); err != nil {
+func (c *CMOSChip) Read(reg memio.Uint8, data memio.UintN) error {
+	if err := c.In(cmosRegPort, &reg); err != nil {
 		return err
 	}
-	return memio.In(cmosDataPort, data)
+	return c.In(cmosDataPort, data)
 }
 
 // Write writes value data into CMOS register reg.
-func Write(reg memio.Uint8, data memio.UintN) error {
-	if err := memio.Out(cmosRegPort, &reg); err != nil {
+func (c *CMOSChip) Write(reg memio.Uint8, data memio.UintN) error {
+	if err := c.Out(cmosRegPort, &reg); err != nil {
 		return err
 	}
-	return memio.Out(cmosDataPort, data)
+	return c.Out(cmosDataPort, data)
+}
+
+// GetCMOS() returns the struct to call Read and Write functions for CMOS
+// associated with the correct functions of memio.In and memio.Out
+func GetCMOS() *CMOSChip {
```

Comment:
Small nit but to follow convention, it would be nice if this were named 
New()
Then the usage would be
cmos.New()
which is a very common way of doing things. 
Also, unless you intend for people to be able to change in and out, following common convention, I would make then non-exported, i.e. in and out.

Reply:
Further, it seems to me that if CMOS implemented the UintN interface, some of this becomes unnecessary. 
Let's say that the following assertion works: 
var _ UintN = &CMOSChip{}

Then the rest of these functions are pretty simple. 

Your mocker, below, would be implemented slightly differently but it would satisfy the type assertion: 

var _  UintN = &CMOSMock {}

So why do this? 
- because of how you are doing it now you are exporting In and Out, which arguably should not be exported
- The "zero value" of a CMOSChip will cause panics, i.e. if someone does this (which is legal)
var c = &CMOSChip{}
c.Out(1) // kaboom!
- the general rule of go is that "zero values" of things should work, and this code violates that rule
but if you do what I'm suggesting, have CMOSChip implement the UintN interface, then this will work. It is more idiomatic. See how, e.g., the zero value of bytes.Buffer "just works": (&bytes.Buffer{}).Bytes() in 
https://play.golang.com/p/9fDKJjcApHb

If you do this, a surprisingly small amount of code needs to change.